packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -39,41 +43,64 @@ export class AtomicSearchInterface {
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() logLevel?: LogLevel;
   @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({reflect: true}) language = 'en'; // TODO: make watchable and update i18next language on change
   @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
+  @Event() ready!: EventEmitter;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
-  private initialized = false;
-  private afterInitializationCallbacks: (() => void)[] = [];
+  private systemsToInitialize = {
+    i18next: false,
+    headless: false,
+  };
 
   componentWillLoad() {
+    this.i18n
+      .use(Backend)
+      .init({
+        debug: this.logLevel === 'debug',
+        lng: this.language,
+        fallbackLng: ['en'],
+        backend: {
+          loadPath: `${getAssetPath('./lang/')}{{lng}}.json`,
+        } as BackendOptions,
+      })
+      .then(() => this.systemInitialized('i18next'));
+
     if (this.sample) {
-      this.initialize(HeadlessEngine.getSampleConfiguration());
+      // Mimics calling the async "initialize" method from the DOM
+      setTimeout(() => {
+        this.initialize(HeadlessEngine.getSampleConfiguration());
+      }, 0);
     }
-
-    this.i18n.use(Backend).init({
-      debug: this.logLevel === 'debug',
-      lng: 'en', // TODO: add watchable "lang" prop
-      fallbackLng: ['en'],
-      backend: {
-        loadPath: `${getAssetPath('./lang/')}{{lng}}.json`,
-      } as BackendOptions,
-    });
   }
 
   disconnectedCallback() {
     this.unsubscribe();
   }
 
-  @Method() async afterInitialization(callback: () => void) {
-    if (this.initialized) {
-      callback();
+  private systemInitialized(system: SystemToInitialize) {
+    this.systemsToInitialize[system] = true;
+    const allSystemsInitialized = Object.keys(this.systemsToInitialize).every(
+      (system) => this.systemsToInitialize[system as SystemToInitialize]
+    );
+
+    if (!allSystemsInitialized) {
       return;
     }
-    this.afterInitializationCallbacks.push(callback);
+
+    this.hangingComponentsInitialization.forEach((event) =>
+      event.detail(this.context)
+    );
+
+    this.initSearchParameterManager();
+    this.ready.emit();
+    this.engine!.dispatch(
```

Comment:
Eventually there should be an option/prop to "load" meaning trigger a search right away. In the case it would be true, should we emit the event after the response? 🤔

Reply:
I would completely remove the "auto trigger query on first load" from inside the search interface/expose no option for this.

With hindsight, I think it caused more problems than it solved in JSUI.

You would, as a end user, always need to decide when to trigger the first query.

We can instead expose a nice method on the search interface to trigger the first query.

```
(async () => {
      await customElements.whenDefined('atomic-search-interface');
      const searchInterface = document.querySelector('#search')
      
      searchInterface.initialize({
        accessToken: 'my_token',
        organizationId: 'my_org',
      });


     // this would be the "classic simple" use case of execute the first query ASAP.
     // Internally, this method would delay for as long as it needs to until all system are initialized
     // returns a promise. 
     searchInterface.executeFirstQuery();

    // could also have custom logic to decide when to execute the first query.
    // for example, say the end user needs to inject custom context in the query, retrieved from a backend system
    retrieveSomethingCustomFromMyBackend().then((context)=> {
            setContext();
            searchInterface.executeFirstQuery();
     })
})();
```


So, to come back to your question, we should emit the event as soon as "everything is technically ready to go as far as atomic is concerned to be able to execute the first query", but we do not decide when that first query is triggered.